packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -104,9 +108,16 @@ export class WorkspaceCommandContribution implements CommandContribution {
                     initialValue: uri.path.base,
                     validate: name => this.validateFileName(name, parent)
                 });
-                dialog.open().then(name =>
-                    this.fileSystem.move(uri.toString(), uri.parent.resolve(name).toString())
-                );
+                dialog.open().then(name => {
+                    this.fileSystem.move(uri.toString(), uri.parent.resolve(name).toString()).then(() => {
+                        this.editorService.renameWidget(uri.parent.resolve(uri.path.base), uri.parent.resolve(name), name);
+                        let event: IUriChangedEvent = { factoryId: uri.parent.resolve(uri.path.base), options: uri.parent.resolve(name), newOptions: name, newLabel: ' ' };
+                        this.uiWatcher.onIUriChangedEmitter.fire(event);
```

Comment:
couple of things
 - you don't need a watcher
 - nothing actually uses that event at the moment ?
 - you still call editorservice rename-widget

It should be that workspace-commands fires the event
and editor service receive it at the very least

Might be other considerations too.. need to think about this more